[Alfred Goh] iP

Alfred-Goh02 opened the pull request.

Alfred-Goh02 committed:
Add List and Mark as Done feature

Implement the ability to list tasks and mark them as done.

The List feature allows users to view all tasks they have added.

The Mark as Done feature enables users to mark individual tasks as
completed.

These features enhance the overall functionality of the Chatbot.

Alfred-Goh02 committed:
Add ToDos, Events, Deadlines

Need ways for user to categorize their tasks.

Users can categorize their tasks into 3 types.

Allow users to track their daily tasks comprehensively.

Alfred-Goh02 committed:
Add Automated Text Ui Test

Simplify debugging process.

Automated test cases help to evaluate code correctness quickly.

Assist developer by reducing debugging effort and time.

C2linaung commented on src/main/java/Poppy.java:
```diff
@@ -0,0 +1,155 @@
+    public static class ToDo extends Task{
+        public ToDo (String description){
+            super(description);
+            System.out.println("[T]" + " [" + super.getStatusIcon()+ "] " + description);
```
I think it would be better to standardise the spacings.. especially between the plus sign

C2linaung commented on src/main/java/Poppy.java:
```diff
@@ -0,0 +1,155 @@
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+
+        public void markAsDone() {
```
using "setDone" is better for function naming. for IP i think is ok to do this since its individual work anyways, but when we doing tP, it would be good to take note for this since we want the code to look like its written by one person.

C2linaung commented on src/main/java/Poppy.java:
```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    public static class Deadline extends Task{
```
Will it be better to put the classes into different files?

C2linaung commented on src/main/java/Poppy.java:
```diff
@@ -0,0 +1,155 @@
+        while (!input.equals("Bye")) {
+            String[] str = input.split(" ", 2 );
+            String command = str[0];
+            switch (command) {
```
I like the use of switch cases here instead of if cases since there are many different conditions.

C2linaung commented on src/main/java/Poppy.java:
```diff
@@ -0,0 +1,155 @@
+        protected String from;
+        public Events (String description, String from){
+            super(description);
+            this.from =from;
```
do take note of the spacings here

C2linaung reviewed (commented):
Generally good job on the codes. I think some suggestions would be to move the codes into different files to  make it easier to organise stuff and also do take note of the spacings

Alfred-Goh02 committed:
Refactor code and improve readability

Alfred-Goh02 committed:
Add Exceptions Handler to Chatbot

Program exits when there is Checked/Unchecked Exceptions.

Added exception handler to catch error so program will not terminate.

try catch blocks can help to catch exception and print message for user.

Alfred-Goh02 committed:
Merge branch 'branch-Level-5' into master

Alfred-Goh02 committed:
Add class for custom exceptions

Alfred-Goh02 committed:
Merge branch 'branch-Level-5' into master

Alfred-Goh02 committed:
Refactor code into packages and modify test file.

Alfred-Goh02 committed:
Merge branch 'branch-A-Packages' into master

Alfred-Goh02 committed:
Add Delete Function to Chatbot

Alfred-Goh02 committed:
Add Save function for ChatBot

Alfred-Goh02 committed:
Merge branch 'branch-Level-6' with master

Alfred-Goh02 committed:
Resolved merge conflicts between master and branch-Level-7

Alfred-Goh02 committed:
Fix Display Issue and Refactor Data File

Alfred-Goh02 committed:
Fix Display Issue for Event Task

okkhoy commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,51 @@
+package commands;
+
+import tasks.*;
```
don't import using wild card importing.

okkhoy commented on src/main/java/commands/Commands.java:
```diff
@@ -0,0 +1,51 @@
+        int index = Integer.parseInt(str[1])- 1;
+        if (list.get(index) == null){
+            System.out.println("Sorry, list number does not exist");
+        }else {
```
spacing issue

okkhoy commented on src/main/java/poppy/Printer.java:
```diff
@@ -0,0 +1,171 @@
+        String command = commandArgs[0];
+        try {
+            switch (command) {
+                case "delete":
+                    delete(taskList, Integer.parseInt(commandArgs[1]));
+                    break;
+                case "mark":
+                    markAsDone(taskList, commandArgs);
+                    break;
+                case "unmark":
+                    markAsNotDone(taskList, commandArgs);
+                    break;
+                case "List":
+                    showList(taskList);
+                    break;
+                case "todo":
```
`case` statements are not following our coding standards

okkhoy commented on src/main/java/poppy/Printer.java:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    public static void processFile(String input, ArrayList<Task> taskList) {
+        String[] commandArgs = input.split(" ", 3);
+        if (commandArgs.length < 3) return; // Ensure there are enough arguments
+
+        String command = commandArgs[0];
+        String status = commandArgs[1];
+        String description = commandArgs[2];
```
can improve SLAP here

okkhoy commented on src/main/java/poppy/Printer.java:
```diff
@@ -0,0 +1,171 @@
+        userInput.close();
+    }
+
+    private static void writeToFile(File file, ArrayList<Task> taskList) throws IOException {
```
All non-trivial methods need header comments

okkhoy reviewed (commented):
I noticed some coding standard violations. Please fix them before the final submission

Alfred-Goh02 committed:
Refactor code to include more classes and improve readability

Alfred-Goh02 committed:
Add Find function to ChatBot

Alfred-Goh02 committed:
Add JavaDoc Comments

Alfred-Goh02 committed:
Merge pull request #1 from Alfred-Goh02/branch-Level-9

Add Find function to ChatBot

Alfred-Goh02 committed:
Merge branch 'master' into A-JavaDoc

Alfred-Goh02 committed:
Merge pull request #3 from Alfred-Goh02/branch-A-JavaDoc

Merge branch-A-JavaDoc into master

Alfred-Goh02 committed:
Edit README

Alfred-Goh02 committed:
Edit Parser and Refactor Code

Alfred-Goh02 committed:
Add to parameter to Events Class

Alfred-Goh02 committed:
Add Datetime formatting for Deadlines

Alfred-Goh02 committed:
Merge pull request #4 from Alfred-Goh02/branch-Level-8

Add Datetime formatting for Deadlines

Alfred-Goh02 committed:
Bug Fixes and Date-Time Format Correction

Alfred-Goh02 committed:
Modify and Update User Guide

Alfred-Goh02 committed:
Bug Fixes
